posthog/hogql/database/database.py

```diff
@@ -115,6 +125,29 @@ def create_hogql_database(team_id: int) -> Database:
     return database
 
 
+def determine_join_function(view):
```

Comment:
Given this is so tightly coupled with the `view`, might be better as a
```python
@property
def join_function
```
on `DataWarehouseViewLink`?